
<span style="font-weight:bold; color:#4CAF50; font-size:28px;">
TASK 1-DATA PIPELINE
DEVELOPMENT<span>

<div style="font-family: Arial, sans-serif; text-align: left; background-color: #f2f2f2; padding: 10px; border-radius: 5px;">
    <span style="font-weight: bold; color: #4CAF50; font-size: 20px;">Author: Bisma yb</span><br>
    <span style="font-weight: bold; color: #FF9800; font-size: 18px;">Domain: Data Science</span><br>
</div>




About this file

This dataset contains 11,768 records of laptops with various hardware specifications and prices. It is designed for predictive modeling,
price estimation, and exploratory data analysis.
The dataset includes real-world values to reflect the factors that influence laptop prices.

In [14]:
import pandas as pd

# Load the dataset
file_path = "C:/Users/Admin/Downloads/laptop_prices.csv"
df = pd.read_csv(file_path)

# Display basic information and first few rows
df.info(), df.head()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11768 entries, 0 to 11767
Data columns (total 11 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Brand                 11768 non-null  object 
 1   Processor             11768 non-null  object 
 2   RAM (GB)              11768 non-null  int64  
 3   Storage               11768 non-null  object 
 4   GPU                   11768 non-null  object 
 5   Screen Size (inch)    11768 non-null  float64
 6   Resolution            11768 non-null  object 
 7   Battery Life (hours)  11768 non-null  float64
 8   Weight (kg)           11768 non-null  float64
 9   Operating System      11768 non-null  object 
 10  Price ($)             11768 non-null  float64
dtypes: float64(4), int64(1), object(6)
memory usage: 1011.4+ KB


(None,
     Brand    Processor  RAM (GB)    Storage                 GPU  \
 0   Apple  AMD Ryzen 3        64  512GB SSD     Nvidia GTX 1650   
 1   Razer  AMD Ryzen 7         4    1TB SSD     Nvidia RTX 3080   
 2    Asus     Intel i5        32    2TB SSD     Nvidia RTX 3060   
 3  Lenovo     Intel i5         4  256GB SSD     Nvidia RTX 3080   
 4   Razer     Intel i3         4  256GB SSD  AMD Radeon RX 6600   
 
    Screen Size (inch) Resolution  Battery Life (hours)  Weight (kg)  \
 0                17.3  2560x1440                   8.9         1.42   
 1                14.0   1366x768                   9.4         2.57   
 2                13.3  3840x2160                   8.5         1.74   
 3                13.3   1366x768                  10.5         3.10   
 4                16.0  3840x2160                   5.7         3.38   
 
   Operating System  Price ($)  
 0          FreeDOS    3997.07  
 1            Linux    1355.78  
 2          FreeDOS    2673.07  
 3          Windo

In [15]:
from sklearn.preprocessing import StandardScaler, OneHotEncoder
from sklearn.impute import SimpleImputer
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline

# Identifying numerical and categorical columns
numerical_features = ["RAM (GB)", "Screen Size (inch)", "Battery Life (hours)", "Weight (kg)", "Price ($)"]
categorical_features = ["Brand", "Processor", "Storage", "GPU", "Resolution", "Operating System"]

# Handling missing values
num_imputer = SimpleImputer(strategy='mean')
cat_imputer = SimpleImputer(strategy='most_frequent')

# Scaling numerical features
scaler = StandardScaler()

# Encoding categorical features
encoder = OneHotEncoder(handle_unknown='ignore')

# Creating pipelines
num_pipeline = Pipeline([
    ('imputer', num_imputer),
    ('scaler', scaler)
])

cat_pipeline = Pipeline([
    ('imputer', cat_imputer),
    ('encoder', encoder)
])

# Applying transformations
preprocessor = ColumnTransformer([
    ('num', num_pipeline, numerical_features),
    ('cat', cat_pipeline, categorical_features)
])



In [17]:
# Convert sparse matrix to dense format
processed_data_dense = processed_data.toarray()

# Convert to DataFrame
processed_df = pd.DataFrame(processed_data_dense, columns=feature_names)

# Save processed dataset
processed_df.to_csv("processed_laptop_prices.csv", index=False)

print("Processed data saved successfully!")


Processed data saved successfully!


Insights from the Data

Price Variation:
The dataset contains laptops with a wide range of prices, from low-budget to high-end gaming/workstations.
Brands like Apple, Razer, and Asus have higher average prices.

Impact of Specifications on Price:

RAM & Storage: More RAM (GB) and larger SSD storage tend to increase price.
Processor & GPU: Laptops with high-end GPUs (e.g., Nvidia RTX 3080) are significantly more expensive.
Screen Resolution: Higher resolutions (4K/3840x2160) increase price.
Feature Engineering Insights:

Storage was initially text-based (e.g., "512GB SSD"). We extracted the numeric storage size.
Categorical Encoding: Transformed Brand, Processor, GPU, etc., into numerical values.

